amr-wind/boundary_conditions/wall_models/WallFunction.cpp

```diff
@@ -116,14 +135,23 @@ void VelWallFunc::wall_model(
                         const amrex::Real vv =
                             vold_arr(i, j, k + idx_offset, 1);
                         const amrex::Real wspd = std::sqrt(uu * uu + vv * vv);
-                        // Dirichlet BC
-                        varr(i, j, k - 1, 2) = 0.0;
-
-                        // Shear stress BC
-                        varr(i, j, k - 1, 0) =
-                            tau.get_shear(uu, wspd) / mu * den(i, j, k);
+                        const amrex::Real xc = problo[0] + (i + 0.5) * dx[0]; 
+			const int z_dx_low = static_cast<int>(std::floor(dx[0] / dx[2])); 
+			const int z_dx_up = z_dx_low + 1;
+			const amrex::Real z_diff = dx[0]  - (z_dx_low + 0.5) * dx[2];
```

Comment:
should it be `dx[2]` if this is a z_diff?

Reply:
I think calculating z_diff this way is incorrect anytime` dx[0] = (a+b/10) * dx[2]`, where `a` and `b` and integers and `b < 5`.

Example: 
```
dx[0] = 2.1 * dx[2]
z_dx_low = 2
z_diff = 2.1*dx[2] - (2+0.5)*dx[2] = -0.4 * dx[2]
```